addons/dragon/functions/fnc_attackProfile_DRAGON.sqf

```diff
@@ -0,0 +1,67 @@
+#include "script_component.hpp"
+/*
+ * Author: Brandon (TCVM) (Code inspired by NouberNou's Dragon Guidance)
+ * Attack profile: Dragon Guidance
+ *
+ * Arguments:
+ * 0: Seeker Target PosASL <ARRAY>
+ * 1: Guidance Arg Array <ARRAY>
+ * 2: Attack Profile State <ARRAY>
+ *
+ * Return Value:
+ * Missile Aim PosASL <ARRAY> - Unused
+ *
+ * Example:
+ * [[1,2,3], [], []] call ace_dragon_fnc_attackProfile_DRAGON;
+ *
+ * Public: No
+ */
+params ["_seekerTargetPos", "_args", "_attackProfileStateParams"];
+_args params ["_firedEH", "", "", "", "_stateParams"];
+_firedEH params ["_shooter","_weapon","","","","","_projectile"];
+_attackProfileStateParams params ["_maxCorrectableDistance", "_wireCut", "_seekerMaxRangeSqr", "_seekerMinRangeSqr", "_wireCutSource", "_lastTime", "_serviceInterval", "_serviceChargeCount", "_serviceChargeAcceleration", "_dragonSpeed"];
+
+private _projectilePos = getPosASL _projectile;
+private _distanceToProjectile = (getPosASL _shooter) vectorDistanceSqr _projectilePos;
+private _retPos = _projectilePos vectorAdd (AGLtoASL (_projectile vectorModelToWorld [0, 50, 0]));
+
+if ((_distanceToProjectile > _seekerMaxRangeSqr) || { _wireCut }) exitWith {
```

Comment:
lazy eval is more expensive than just checking the local variable directly.
Either swap it around that the expensive condition is in lazy eval, or just get rid of the `{}`